### DISCUSSION TITLE
License/Upstream to ESPHome

### DISCUSSION THREAD

**Discussion by brotherdust** [TheRealWaldo/AMG8833-ESPHOME#11] [Ideas]
Hello!
Have you considered, or, are interested in, turning this into a component and submitting it to ESPHome? 

If not, I have some experience writing them and would like to do so; but I'm cautious about doing so with code that has no license statement. Would you mind setting your license so I know what is OK to do with your code?

Thanks much!

**Comment by TheRealWaldo** [comment]
@brotherdust definitely interested in getting the AMG8833 and a few other sensors into ESPHome, and I have been wanting to do so and did start a project for it; would be happy to work with you to do get it working!

**Comment by TheRealWaldo** [comment]
I've just pushed the work towards converting this into an external component so far to https://github.com/TheRealWaldo/AMG8833-ESPHOME/tree/external_component

There are a few things that I still need to complete.

I want to figure out if it's okay to immediately publish the state from the sensors rather than wait for polling, and this would be best for both the camera and person detection.  I think it's as simple as adding the publish in the loop instead of the update, but I need to experiment and confirm if there is any precedent to doing that.

If not, I'll have to do some refactoring to break those sensors into individual classes.

The `text_sensor` I'm using for the camera needs some love, as does the root `sensor` component.

Lastly, I've got to port the library in since ESPHome does not allow for external libraries.

**Comment by rrooggiieerr** [comment]
Hi @TheRealWaldo! I'm looking into using your AMG8833 implementation into my ESPHome project. What's the status on the external component?
